pkg/scheduler/objects/application_state.go

```diff
@@ -142,6 +143,9 @@ func NewAppState() *fsm.FSM {
 			},
 			fmt.Sprintf("enter_%s", Starting.String()): func(event *fsm.Event) {
 				app := event.Args[0].(*Application) //nolint:errcheck
+				if app.queue != nil {
+					app.queue.runningApps++
```

Comment:
We have to update the counter in the parent as well, all the way up to root.

So we need an `incrementRunningApps()` method like:
```
func (sq *Queue) IncrementRunningApps() {
	sq.Lock()
	defer sq.Unlock()
	sq.runningApps++
	if sq.parent != nil {
		sq.parent.IncrementRunningApps()		
	}
}
```